src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,266 @@
+import {horizontalScale, verticalScale} from '../../utils';
+import {Label} from '../Label';
+import {TextInputProps} from './TextInput.types';
+
+export const TextInput = (props: TextInputProps) => {
+  const {
+    mode = 'default',
+    onChangeText,
+    hint,
+    style,
+    placeholder,
+    hintStyle,
+    error,
+    errorMessage,
+    errorMessageStyle,
+    label,
+    labelStyle,
+    editable,
+    inputStyle,
+    left,
+    right,
+    value,
+    ...restProps
+  } = props;
+  const [focused, setFocused] = useState(false);
+  const {colors} = useThemeColor();
+
+  const getStyle = useMemo(() => {
+    if (mode === 'outline') {
+      return styles.outline;
+    } else if (mode === 'border-less') {
+      return styles.borderLess;
+    }
+    return;
+  }, [mode]);
+
+  const getDisabledStyle = useMemo(() => {
+    if (!editable) {
+      if (mode === 'outline') {
+        return [
+          styles.outlineDisable,
+          {
+            borderColor: colors.placeholder,
+            backgroundColor: colors.disabled,
+          },
+        ];
+      } else if (mode === 'default') {
+        return [styles.defaultDisable, {borderBottomColor: colors.placeholder}];
+      }
+    } else {
+      return;
+    }
+  }, [colors.disabled, colors.placeholder, editable, mode]);
+
+  const floatingLabel = useRef(new Animated.Value(0)).current;
+
+  const yInterpolate = floatingLabel.interpolate({
+    inputRange: [0, 1],
+    outputRange: [4, -15],
+  });
+
+  const animation = {
+    transform: [
+      {
+        translateY: yInterpolate,
+      },
+    ],
+  };
+
+  const moveLabelTop = () => {
+    Animated.timing(floatingLabel, {
+      toValue: 1,
+      duration: 200,
+      useNativeDriver: true,
+    }).start();
+  };
+
+  const moveLabelDown = () => {
+    Animated.timing(floatingLabel, {
+      toValue: 0,
+      duration: 200,
+      useNativeDriver: true,
+    }).start();
+  };
+
+  const onFocusHandler = () => {
+    moveLabelTop();
+  };
+
+  const onBlurHandler = (e: string) => {
+    if (e?.length === 0 || value?.length === 0) {
+      moveLabelDown();
+    }
+  };
+
+  const renderAnimatedLabel = () => {
+    const animatedLabelStyle = [
+      animation,
+      styles.animatedLabel,
+      mode === 'outline'
+        ? {
+            backgroundColor: colors.background,
+          }
+        : null,
+    ];
+
+    const renderLabelStyle = [
+      {
+        backgroundColor:
+          !editable && mode === 'outline' ? colors.disabled : colors.background,
+        color: colors.text,
+      },
+      fontStyles.b3_Text_Italic,
+      focused ? [fontStyles.b3_Text_SemiBold, labelStyle] : null,
+    ];
+
+    if (label) {
+      return (
+        <Animated.View style={animatedLabelStyle}>
+          <Label style={renderLabelStyle} label={label} />
+        </Animated.View>
+      );
+    }
+    return null;
+  };
+
+  const onFocus = () => {
+    onFocusHandler();
+    setFocused(true);
+    restProps.onFocus;
```

Comment:
Will this work? have you tested it?

Reply:
Yeah it is working as expected. I tried with alerts and also console logs and they correctly showed up

Reply:
You are not even executing the function. How is it working?